core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala

```diff
@@ -704,6 +707,43 @@ class MasterSuite extends SparkFunSuite
   private def getState(master: Master): RecoveryState.Value = {
     master.invokePrivate(_state())
   }
+
+  test("Total cores is not divisible by cores per executor") {
```

Comment:
My original thinking is that if it is possible to test wether the `if` branch is executed or not using `mock` and `verify`. But looks like it is not easy to test. Can you please investigate.